pkg/apis/core/validation/validation.go

```diff
@@ -2907,51 +2907,49 @@ func validatePullPolicy(policy core.PullPolicy, fldPath *field.Path) field.Error
 	return allErrors
 }
 
+// validateEphemeralContainers is called by pod spec and template validation to validate the list of ephemeral containers.
+// Note that this is called for pod template even though ephemeral containers aren't allowed in pod templates.
```

Comment:
Should we fix the fact that this is run for templates?  It might require more plumbing, in a different PR?